code/drasil-code/lib/Language/Drasil/CodeSpec.hs

```diff
@@ -93,11 +99,12 @@ codeSpec SI {_sys         = sys
            , _sysinfodb   = db} chs ms =
   let n = programName sys
       inputs' = map quantvar ins
-      const' = map qtov (filter ((`Map.notMember` conceptMatch chs) . (^. uid))
+      const' = map qtov (filter ((`Map.notMember` conceptMatch (maps chs)) . (^. uid))
         cnsts)
       derived = map qtov $ getDerivedInputs ddefs inputs' const' db
       rels = (map qtoc (getEqModQdsFromIm ims ++ mapMaybe qdEFromDD ddefs) \\ derived)
-        ++ map odeDef (odes chs)
+        ++ if isNothing ode then [] else map odeDef (odeInfo $ fromJust ode)
```

Comment:
here you should add an auxilliary function that does the work, and can do it by pattern-matching on `ode`.

Reply:
Removed the if statement. An auxiliary function was created with pattern matching.